### PR TITLE
doc: document that 'openssl rand' is cryptographically secure

### PR DESCRIPTION
**Pull request by mspncp**
Alternative to #11202.

### CODE REVIEW

File: doc/man3/RAND_bytes.pod [hunk 1]
```diff
@@ -44,10 +44,21 @@ ignored.
 
 =head1 NOTES
 
-Always check the error return value of RAND_bytes() and
-RAND_priv_bytes() and do not take randomness for granted: an error occurs
-if the CSPRNG has not been seeded with enough randomness to ensure an
-unpredictable byte sequence.
+By default, the OpenSSL CSPRNG supports a security level of 256 bits, provided it
+was able to seed itself with 256 bits of entropy from a trusted entropy source.
```
Comment (paulidale) [inline doc/man3/RAND_bytes.pod]: `d/itself with 256 bits of entropy/`

It grabs 384 bits, so best to not say how much :)

File: doc/man3/RAND_bytes.pod [hunk 2]
```diff
@@ -44,10 +44,21 @@ ignored.
...
+
+If the entropy source fails or is not available, the CSPRNG will enter an
+error state and refuse to generate random bytes. For that reason, it is important
+to always check the error return value of RAND_bytes() and RAND_priv_bytes() and
+not take randomness for granted.
+
+On other more exotic platforms, there might not be a trusted entropy source available
+or OpenSSL might have been explicitly configured to use different entropy sources.
+If you are in doubt about the quality of the entropy source, don't hesitate to ask
+your os vendor or post a question on GitHub or the openssl-users mailing list.
```
Comment (paulidale) [inline doc/man3/RAND_bytes.pod]: `s/ os / OS /` or better `s/ os / operating system /`

File: doc/man3/RAND_bytes.pod [hunk 3]
```diff
@@ -44,10 +44,21 @@ ignored.
...
+On all major platforms supported by OpenSSL (including the Unix-like platforms
+and Windows), OpenSSL is configured to automatically seed the CSPRNG on first use
+using the operating systems's random generator.
+
+If the entropy source fails or is not available, the CSPRNG will enter an
+error state and refuse to generate random bytes. For that reason, it is important
+to always check the error return value of RAND_bytes() and RAND_priv_bytes() and
+not take randomness for granted.
+
+On other more exotic platforms, there might not be a trusted entropy source available
```
Comment (paulidale) [inline doc/man3/RAND_bytes.pod]: Consider perhaps: `s/more exotic/other/`

File: doc/man3/RAND_bytes.pod (hunk 1 above)
Comment (mspncp) [reply doc/man3/RAND_bytes.pod]: Fixed in 68d8704e8c2ec5c92ce769592babcb74e31c27e3.

File: doc/man3/RAND_bytes.pod (hunk 3 above)
Comment (mspncp) [reply doc/man3/RAND_bytes.pod]: Fixed in a6eefe61ae9c21e8745162f191ad0672249bf7b2.

File: doc/man3/RAND_bytes.pod (hunk 2 above)
Comment (mspncp) [reply doc/man3/RAND_bytes.pod]: Done, see 9ea4fcaeb94714649bf9ef72cde1cc7aba26f471.

**Comment by mspncp** [comment]
Note: commit 7c1748bf80ef57b9223b15efd065858ad1799647 only adds 'Fixes #11201' to the commit message.

**Comment by mspncp** [comment]
Oh, and do I get your approval for 1.1.1 as well?

**Comment by paulidale** [comment]
Yes, 1.1.1 is good too.

**Comment by mspncp** [comment]
Merged to master as 88398d2a358fe41e33c61ac02f23ffaeacddcff0. Cherry-pick to 1.1.1 had conflicts which seem to be a little bit more than just a simple tree conflict.

    doc/man1/rand.pod -> doc/man1/openssl-rand.pod -> doc/man1/openssl-rand.pod.in

It needs some manual adjustment which is probably straight forward, but due to lack of time can't be done immediately. So I leave this pr open as a reminder until I find the time to open a separate pr for 1.1.1.

**Comment by romen** [comment]
This was merged even if Travis errors were relevant: `make doc-nits` fails.

In `master` this was addressed with #11231 (it almost looks intentional that ##11321 fixes #11213 !), but we should fix it in `1.1.1` as well!

**Comment by mspncp** [comment]
> This was merged even if Travis errors were relevant: `make doc-nits` fails.
> 
> In `master` this was addressed with #11231 (it almost looks intentional that ##11321 fixes #11213 !), but we should fix it in `1.1.1` as well!

Thanks for the heads-up and sorry that this bug slipped in unnoticed. Since this pull request still needs to be backported (the cherry-pick was not clean), I will create a separate pull request without doc-nits failures.

**Comment by mspncp** [comment]
Backport is now in #11251.

**Pull request closed without merging**